src/transform/ReferenceCollection.js

```diff
@@ -0,0 +1,253 @@
+import ElementUtilities from './ElementUtilities'
+import NodeUtilities from './NodeUtilities'
+import Polyfill from './Polyfill'
+
+const REFERENCE_SELECTOR = '.reference, .mw-ref'
+const CITE_HASH_PREFIX = '#cite_note'
+
+/**
+ * Is Citation.
+ * @param {!string} href
+ * @return {!boolean}
+ */
+const isCitation = href => href.indexOf(CITE_HASH_PREFIX) > -1
+
+/**
+ * Gets first child anchor.
+ * @param {!HTMLElement} element
+ * @return {?HTMLAnchorElement}
+ */
+const getFirstChildAnchor = element => element.querySelector('A')
```

Comment:
Heya!

I wanted to share my perspective on replacing comments with functions as you've done here.

In general, I think it's great practice to use good naming on our symbols and, of course, abstraction for the win. However, I probably wouldn't write a comment for querying the first anchor of a DOM subtree since it's so commonplace as to be un-noteworthy and the abstraction here does nothing. Function calls may now read more like English but they also obscure the underlying simplicity of what's actually happening here with a new custom DSL and greatly inflate the verbosity of the code which is more of a decrease to readability than an increase.

Consider what happens also when the second child anchor is needed in some new patch. Do we make a new function that sits alongside getFirstChildAnchor() called getSecondChildAnchor()? That seems bad enough but when the third anchor is needed, suddenly we might wonder if getNChildAnchor() should be added which takes an argument, n. Now we have three functions that all boil down to querySelector / querySelectorAll and we still need to pass an argument.

I think we both feel very comfortable working with querySelector / querySelectorAll so my opinion is that we should favor their direct usage unless we're doing something extra that we want to capture in a function.

Reply:
Ya this was overly abstracted. I removed it. Good catch :)